packages/pwa-kit-runtime/src/ssr/server/build-remote-server.js

```diff
@@ -473,6 +486,213 @@ export const RemoteServerFactory = {
+                        result += char
+                    }
+                } else if (char === '?' && (inParameter || parenDepth > 0)) {
+                    // This ? is part of a parameter or group, keep as is
+                    result += char
+                    if (inParameter) {
+                        inParameter = false
+                    }
+                } else {
+                    result += char
+                }
+                i++
+            }
+
+            return result
+        }
+
+        const convertExpressRouteToRegex = (routePattern) => {
+            if (!routePattern) return null
+            if (routePattern instanceof RegExp) return routePattern
+            if (typeof routePattern !== 'string') return null
+
+            try {
+                // First, convert Express wildcards to path-to-regexp syntax
+                // Express uses * for wildcards, path-to-regexp uses (.*)
+                // We need to do this before handling optional (?) characters
+                let processedPattern = routePattern.replace(/\*/g, '(.*)')
+
+                // Convert any ? in the express route to something path-to-regexp can process
+                processedPattern = convertExpressOptionalChars(processedPattern)
+
+                // Ensure we have a valid string
+                if (!processedPattern || typeof processedPattern !== 'string') {
+                    throw new Error(`Invalid processed pattern: ${processedPattern}`)
+                }
+
+                // Use path-to-regexp to convert the route pattern to a regex
+                return pathToRegexp(processedPattern)
+            } catch (error) {
+                throw new Error(`Invalid route pattern: ${routePattern}`)
+            }
+        }
+
+        // Initializes a cache of regexes that correspond to the registered express routes
+        const initializeExpressRouteRegexes = () => {
+            const expressRoutes = app._router.stack
+                // Specifically omit the generic wildcard from the express routes we want to
+                // remove the base path from since it is mapped to the app render. Routes
+                // within the render are handled by React Router
+                .filter((layer) => layer.route && layer.route.path && layer.route.path !== '*')
+                .map((layer) => layer.route.path)
+
+            expressRouteRegexes = expressRoutes.map((route) => convertExpressRouteToRegex(route))
+        }
+
+        /**
+         * Very early request processing.
+         *
+         * If the server receives a request containing the base path, remove it before allowing it through
+         *
+         * @param req {express.req} the incoming request - modified in-place
+         * @private
+         */
+        const removeBasePathMiddleware = (req, res, next) => {
+            // Scope base path removal to /mobify routes and routes defined by the express app (ie. worker.js)
```

Comment:
having comments help a lot, but I think it is way to hard to read the code now. 🤔 . Maybe shorten comments a bit, or move them to the jsdoc to explain what this func can do, provide example is a good way to do it